### DISCUSSION TITLE
Add Todoist task with the current line

### DISCUSSION THREAD

**Discussion by daviddelven** [jamiebrynes7/obsidian-todoist-plugin#173] [Q&A]
Hello, 

Is there any plan to add the command enabling creating a task with the current line, besides the current note?
I currently use the combination of your plugin and the Note Refactor one to create Tasks on the fly, but the extra NR step creates some friction. 

Let us know. 

Thanks for your contribution. 

David.

**Comment by g12r** [comment]
Out of curiosity and on a major lag, how are you using Note Refactor in this scenario? I was looking at adding the ability for the plugin to determine if you are in a normal markdown task in the Obsidian editor and, if so, to preload the modal you get for "Todoist Plugin: Add todoist task" using the markdown task text. I don't use tags or dates in my markdown tasks, but I suspect you could also parse those out as well. Really, all I'm saving here is copying and pasting in the task note, but command > Enter would be a decent simplification.

What I haven't figured out is do I ever want to delete the markdown task once it's been successfully posted to todoist? I might do the first step and see if I end up wanting a copy or a move in practice. 

Added: it already works as-is with the current editor selection, so it's not even much of a gain to have it auto-select the current task (if nothing is selected, just a cursor position). That said, I could also so just hotkeying to trigger a silent post of the current task content, selected or inferred, being faster. It should depend a good bit on how often you want to create in Obsidian and copy to Todoist, which isn't my dominant workflow.
